test/SILGen/other-entry-point-function-name.swift

```diff
@@ -0,0 +1,9 @@
+// RUN: %target-run-simple-swift(-Xfrontend -entry-point-function-name -Xfrontend foobar -Xlinker -e -Xlinker _foobar) | %FileCheck %s
```

Comment:
This probably isn't a great test because `-e` controls the name of `start`, not the name of `main`, and we still want dynamic linker and C runtime initialization to happen before we enter `main`. Maybe link against a little stub .o file that provides a `main` and forwards to `_foobar`? That should also be more portable.

Reply:
How's https://github.com/apple/swift/pull/35595/files#diff-a8bcdd22afab25318feddb2667ef722aa57064b5fc9c8bd406de22ad34c1e162R4 for the source for the .o file that provides `main` and forwards to `_foobar`?

Reply:
I think that `envp` is the third argument, at least on some platforms?

I wonder how easy it will be to create a portable stub that works on all platforms.  It shouldn't be super-difficult, but there might have to be some on-going engineering due to subtleties.

Reply:
Swift's generated `main` doesn't use any arguments other than `argc` or `argv`, so I think this should be OK.

Reply:
`main` on Windows is definitely more interesting:

```c
int main(int argc, char *argv);
int wmain(int argc, WCHAR *argv);
```

Reply:
@compnerd Which of those entry points are currently used for Swift's generated `main`?

Reply:
@jckarter Were there other changes to the test?  It is using the .o file generated from the source linked above to forward main to _foobar (and runs on all platforms).  https://github.com/apple/swift/pull/35595/files#diff-a522629ea40edb88675159bc50bd75bae5a0cf4ab8038c781df4790a270af888R3

Reply:
@compnerd But we don't ever generate anything other than the standard `main` today, do we? This should be fine for now, since we'd probably have some opt-in way to use alternative entry points in the future.